core/src/main/java/io/questdb/cairo/map/MapFactory.java

```diff
@@ -108,14 +68,16 @@ private static Map createMap(
         throw CairoException.critical(0).put("unknown map type: ").put(mapType);
     }
 
-    private static Map createMap(
-            boolean smallMap,
+    /**
+     * Creates a Map pre-allocated to a small capacity to be used in SAMPLE BY, GROUP BY queries, but not only.
+     */
+    public static Map createMap(
             CairoConfiguration configuration,
             @Transient @NotNull ColumnTypes keyTypes,
             @Transient @NotNull ColumnTypes valueTypes
     ) {
-        final int keyCapacity = smallMap ? configuration.getSqlSmallMapKeyCapacity() : configuration.getSqlMapKeyCapacity();
```

Comment:
We should mark `cairo.sql.map.key.capacity` and `cairo.sql.map.page.size` as deprecated in `PropServerConfiguration`. Also, we could remove the methods from the `CairoConfiguration` interface - this would be a breaking change for embedders, but it should be trivial to fix on their side.

Reply:
@puzpuzpuz how would you propose to depricate config?

Reply:
We already have everything in-place for that. See https://github.com/questdb/questdb/blob/3887e660e4d169801d14d694748116c33ab06cba/core/src/main/java/io/questdb/PropServerConfiguration.java#L1481-L1538

Reply:
fixed in https://github.com/questdb/questdb/pull/3399/commits/e168d688c5d60018663d2c4c34aa90ab20f6f7b8